File: backend/src/main/java/com/btalk/dto/CallSignal.java
package com.btalk.dto;

import lombok.Data;
import java.util.List;
import com.btalk.constants.SignalType;

@Data
public class CallSignal {
    private String conversationId;
    private String callerId;
    private String recipientId; // For private calls
    private List<String> participants; // For group calls (changed from Set to List to match frontend)
    private SignalType type; // "offer", "answer", "candidate", "ringing", "hangup"
    private Object payload; // SDP offer/answer or ICE candidate
    private String callId; // Add callId field to match frontend
}

File: frontend/src/app/models/call.signal.ts
import { SignalType } from "./call.enum";

export interface CallSignal {
  conversationId?: string; // UUID as string
  callerId: string; // UUID as string
  recipientId?: string; // UUID as string, for private calls
  participants?: string[]; // UUIDs as strings, for group calls
  type: SignalType;
  payload: any; // SDP offer/answer or ICE candidate
  callId?: string; // Add if needed for call tracking
}